plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1088,11 +1107,137 @@ def run(
+               to the detector plane coordinates.
+           * - ``"xloc0"``
+             - `~numpy.ndarray`, [``nparticles``,]
+             - The x-coordinate location where each particle would have
+               hit the detector plane if the grid fields were zero, in
+               meters. Useful for calculating the source profile.
+           * - ``"yloc0"``
+             - `~numpy.ndarray`, [``nparticles``,]
+             - The y-coordinate location where each particle would have
+               hit the detector plane if the grid fields were zero, in
+               meters. Useful for calculating the source profile.
+           * - ``"v0"``
+             - `~numpy.ndarray`, [``nparticles``, 3]
+             - The velocity of each particle when it hit the detector
+               plan if the grid fields were zero, in meters per second.
+               The velocity is in a coordinate system relative to the
+               detector plane. The components are [normal, horizontal,
+               vertical] relative to the detector plane coordinates.
+
+        """
+
+        if not self._has_run:
+            raise RuntimeError(
+                "The simulation must be run before a results "
+                "dictionary can be created."
+            )
+
+        # Determine locations of points in the detector plane using unit
+        # vectors
+        xloc = np.dot(self.x - self.detector, self.det_hdir)
+        yloc = np.dot(self.x - self.detector, self.det_vdir)
+
+        x0loc = np.dot(self.x0 - self.detector, self.det_hdir)
+        y0loc = np.dot(self.x0 - self.detector, self.det_vdir)
+
+        # Determine the velocity components of each particle in the
+        # coordinate frame of the detector, eg. the components of v are
+        # now [normal, det_hdir, det_vdir]
+        v = np.zeros(self.v.shape)
+        v[:, 0] = np.dot(self.v, self.det_n)
+        v[:, 1] = np.dot(self.v, self.det_hdir)
+        v[:, 2] = np.dot(self.v, self.det_vdir)
+
+        v0 = np.zeros(self.v.shape)
+        v0[:, 0] = np.dot(self.v_init, self.det_n)
+        v0[:, 1] = np.dot(self.v_init, self.det_hdir)
+        v0[:, 2] = np.dot(self.v_init, self.det_vdir)
+
+        # Store output values in a dictionary
+        result_dict = dict(
+            source=self.source,
+            detector=self.detector,
+            mag=self.mag,
+            nparticles=self.nparticles,
+            max_deflection=self.max_deflection,
+            x=xloc,
+            y=yloc,
+            v=v,
+            x0=x0loc,
+            y0=y0loc,
+            v0=v0,
+        )
+
+        return result_dict
+
+    def save_results(self, path):
+        """
+        Save the output dictionary containing the final particle positions in
+        the detector plane into a `numpy.npz` file for later analysis.
+
+        Parameters
+        ----------
+
+        path : str or `os.path`
+            The path to save the output file.
```

Comment:
```suggestion
        path : `str` or `os.path`
            Either the filename (string) or an open file (file-like object)
            where the data will be saved. If file is a string or a Path,
            the ``.npz`` extension will be appended to the filename if
            it is not already there.
```

Since you're leveraging `np.savez` just use there definition for the argument to remain consistent.